lib/auth/auth_with_roles.go

```diff
@@ -1193,7 +1192,7 @@ func (a *AuthWithRoles) DeleteAllRemoteClusters() error {
 // signed certificate if sucessfull.
 func (a *AuthWithRoles) ProcessKubeCSR(req KubeCSR) (*KubeCSRResponse, error) {
 	// limits the requests types to proxies to make it harder to break
-	if !a.checker.HasRole(string(teleport.RoleProxy)) {
+	if !a.isBuiltinRole || !a.checker.HasRole(string(teleport.RoleProxy)) {
```

Comment:
let's have a helper method:

```
func (a *AuthWithRoles) hasBuiltinRole(name string) bool {
   return a.checker.IsBuiltin() && a.checker.HasRole(string(name))
}
```

and replace all occurrences to use it everywhere, so we do less errors.